<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>821</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QScrollArea" name="scrollArea_img1">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>30</y>
      <width>241</width>
      <height>221</height>
     </rect>
    </property>
    <property name="verticalScrollBarPolicy">
     <enum>Qt::ScrollBarAlwaysOn</enum>
    </property>
    <property name="horizontalScrollBarPolicy">
     <enum>Qt::ScrollBarAlwaysOn</enum>
    </property>
    <property name="widgetResizable">
     <bool>true</bool>
    </property>
    <widget class="QWidget" name="scrollAreaWidgetContents">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>225</width>
       <height>205</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="widget_mode" native="true">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>471</width>
      <height>30</height>
     </rect>
    </property>
    <widget class="QPushButton" name="pushButton_addline">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>5</y>
       <width>71</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>Add line</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_mergelines">
     <property name="geometry">
      <rect>
       <x>180</x>
       <y>5</y>
       <width>111</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>Merge lines V/H</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_joincorner">
     <property name="geometry">
      <rect>
       <x>300</x>
       <y>5</y>
       <width>91</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>Join corner</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_removeline">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>5</y>
       <width>89</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>Remove line</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_show3D">
     <property name="geometry">
      <rect>
       <x>400</x>
       <y>5</y>
       <width>71</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>show 3D</string>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="widget_control" native="true">
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>140</y>
      <width>261</width>
      <height>411</height>
     </rect>
    </property>
    <property name="autoFillBackground">
     <bool>true</bool>
    </property>
    <widget class="QTreeWidget" name="lines_treeWidget">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>0</y>
       <width>241</width>
       <height>311</height>
      </rect>
     </property>
     <property name="indentation">
      <number>20</number>
     </property>
     <property name="rootIsDecorated">
      <bool>false</bool>
     </property>
     <property name="itemsExpandable">
      <bool>false</bool>
     </property>
     <property name="headerHidden">
      <bool>true</bool>
     </property>
     <property name="expandsOnDoubleClick">
      <bool>false</bool>
     </property>
     <attribute name="headerVisible">
      <bool>false</bool>
     </attribute>
     <attribute name="headerCascadingSectionResizes">
      <bool>false</bool>
     </attribute>
     <attribute name="headerDefaultSectionSize">
      <number>20</number>
     </attribute>
     <attribute name="headerMinimumSectionSize">
      <number>20</number>
     </attribute>
     <attribute name="headerStretchLastSection">
      <bool>true</bool>
     </attribute>
     <column>
      <property name="text">
       <string>#</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>x1</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>y1</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>x2</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>y2</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>vhp</string>
      </property>
     </column>
    </widget>
    <widget class="QWidget" name="widget_bt_ls" native="true">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>310</y>
       <width>261</width>
       <height>101</height>
      </rect>
     </property>
     <widget class="QPushButton" name="pushButton_savelines">
      <property name="geometry">
       <rect>
        <x>130</x>
        <y>0</y>
        <width>61</width>
        <height>25</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Save lines</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_loadlines">
      <property name="geometry">
       <rect>
        <x>190</x>
        <y>0</y>
        <width>61</width>
        <height>25</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Load lines</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_resetlines">
      <property name="geometry">
       <rect>
        <x>190</x>
        <y>25</y>
        <width>61</width>
        <height>25</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
      <property name="text">
       <string>reset</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_saveimage">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>0</y>
        <width>121</width>
        <height>25</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Save image of lines</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_whitebg">
      <property name="geometry">
       <rect>
        <x>130</x>
        <y>25</y>
        <width>61</width>
        <height>25</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
      <property name="text">
       <string>white bg</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_showlen">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>50</y>
        <width>121</width>
        <height>25</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Show wall length</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_savepixmap">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>25</y>
        <width>121</width>
        <height>25</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Save show pixmap</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_testlight">
      <property name="geometry">
       <rect>
        <x>190</x>
        <y>50</y>
        <width>61</width>
        <height>25</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
      <property name="text">
       <string>3D test</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_imgsize">
      <property name="geometry">
       <rect>
        <x>16</x>
        <y>80</y>
        <width>231</width>
        <height>20</height>
       </rect>
      </property>
      <property name="layoutDirection">
       <enum>Qt::LeftToRight</enum>
      </property>
      <property name="text">
       <string>img size</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QScrollArea" name="scrollArea_img2">
    <property name="geometry">
     <rect>
      <x>240</x>
      <y>30</y>
      <width>241</width>
      <height>221</height>
     </rect>
    </property>
    <property name="verticalScrollBarPolicy">
     <enum>Qt::ScrollBarAlwaysOn</enum>
    </property>
    <property name="horizontalScrollBarPolicy">
     <enum>Qt::ScrollBarAlwaysOn</enum>
    </property>
    <property name="widgetResizable">
     <bool>true</bool>
    </property>
    <widget class="QWidget" name="scrollAreaWidgetContents_2">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>225</width>
       <height>205</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="QLabel" name="label_img1">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>250</y>
      <width>151</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>label_img1</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_img2">
    <property name="geometry">
     <rect>
      <x>240</x>
      <y>250</y>
      <width>141</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>label_img2</string>
    </property>
   </widget>
   <widget class="QTabWidget" name="tabWidget_hough_custom">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>0</y>
      <width>241</width>
      <height>141</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>2</number>
    </property>
    <widget class="QWidget" name="Hough">
     <attribute name="title">
      <string>Hough</string>
     </attribute>
     <widget class="QLineEdit" name="lineEdit_maxgap">
      <property name="geometry">
       <rect>
        <x>205</x>
        <y>60</y>
        <width>31</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>0</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineEdit_thres">
      <property name="geometry">
       <rect>
        <x>205</x>
        <y>20</y>
        <width>31</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>0</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="label_pixel">
      <property name="geometry">
       <rect>
        <x>2</x>
        <y>0</y>
        <width>67</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>pixel</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
      </property>
     </widget>
     <widget class="QSlider" name="HSlider_maxgap">
      <property name="geometry">
       <rect>
        <x>60</x>
        <y>60</y>
        <width>141</width>
        <height>16</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QLabel" name="label_maxgap">
      <property name="geometry">
       <rect>
        <x>2</x>
        <y>60</y>
        <width>67</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>maxgap</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
      </property>
     </widget>
     <widget class="QSlider" name="HSlider_pixel">
      <property name="geometry">
       <rect>
        <x>60</x>
        <y>0</y>
        <width>141</width>
        <height>16</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QSlider" name="HSlider_thres">
      <property name="geometry">
       <rect>
        <x>60</x>
        <y>20</y>
        <width>141</width>
        <height>16</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QLabel" name="label_minlen">
      <property name="geometry">
       <rect>
        <x>2</x>
        <y>40</y>
        <width>71</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>min len</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineEdit_minlen">
      <property name="geometry">
       <rect>
        <x>205</x>
        <y>40</y>
        <width>31</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>0</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineEdit_pixel">
      <property name="geometry">
       <rect>
        <x>205</x>
        <y>0</y>
        <width>31</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>0</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="label_thres">
      <property name="geometry">
       <rect>
        <x>2</x>
        <y>20</y>
        <width>67</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>thres</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
      </property>
     </widget>
     <widget class="QSlider" name="HSlider_minlen">
      <property name="geometry">
       <rect>
        <x>60</x>
        <y>40</y>
        <width>141</width>
        <height>16</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_houghcal">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>80</y>
        <width>81</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>hough cal</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="Custom">
     <attribute name="title">
      <string>Custom</string>
     </attribute>
     <widget class="QPushButton" name="pushButton_cal_sp_line">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>80</y>
        <width>61</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>cal lines</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineEdit_minlen_custom">
      <property name="geometry">
       <rect>
        <x>205</x>
        <y>20</y>
        <width>31</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>0</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineEdit_maxgap_custom">
      <property name="geometry">
       <rect>
        <x>205</x>
        <y>40</y>
        <width>31</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>0</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="label_minlen_custom">
      <property name="geometry">
       <rect>
        <x>2</x>
        <y>20</y>
        <width>71</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>min len</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="label_maxgap_custom">
      <property name="geometry">
       <rect>
        <x>2</x>
        <y>40</y>
        <width>67</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>maxgap</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
      </property>
     </widget>
     <widget class="QSlider" name="HSlider_maxgap_custom">
      <property name="geometry">
       <rect>
        <x>60</x>
        <y>40</y>
        <width>141</width>
        <height>16</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QSlider" name="HSlider_minlen_custom">
      <property name="geometry">
       <rect>
        <x>60</x>
        <y>20</y>
        <width>141</width>
        <height>16</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="Merge">
     <attribute name="title">
      <string>Merge</string>
     </attribute>
     <widget class="QPushButton" name="pushButton_merge_corner">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>80</y>
        <width>81</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>m_corners</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_merge_sp_line">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>80</y>
        <width>81</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>mergelines</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineEdit_nearby">
      <property name="geometry">
       <rect>
        <x>205</x>
        <y>0</y>
        <width>31</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>0</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="label_maxspace">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>20</y>
        <width>71</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>maxspace</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
      </property>
     </widget>
     <widget class="QSlider" name="HSlider_cornerrad">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>50</y>
        <width>131</width>
        <height>20</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QLabel" name="label_nearby">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>67</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>near by</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineEdit_cornerrad">
      <property name="geometry">
       <rect>
        <x>205</x>
        <y>50</y>
        <width>31</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>0</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QSlider" name="HSlider_maxspace">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>20</y>
        <width>131</width>
        <height>20</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineEdit_maxspace">
      <property name="geometry">
       <rect>
        <x>205</x>
        <y>20</y>
        <width>31</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>0</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QSlider" name="HSlider_nearby">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>0</y>
        <width>131</width>
        <height>20</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QLabel" name="label_cornerrad">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>50</y>
        <width>71</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>corner rad</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>821</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuDisplay">
    <property name="title">
     <string>Display</string>
    </property>
    <addaction name="actionLoad_image"/>
    <addaction name="actionRaw_image"/>
    <addaction name="action2D_layout"/>
    <addaction name="actionReset_layout"/>
    <addaction name="actionPredict_Layout"/>
   </widget>
   <widget class="QMenu" name="menuBatchOp">
    <property name="title">
     <string>BatchOp(loop)</string>
    </property>
    <addaction name="actioncalline_save"/>
    <addaction name="actionmergeline_save"/>
    <addaction name="actionjoincorner_save"/>
    <addaction name="actionlayout_textsize_save"/>
   </widget>
   <addaction name="menuDisplay"/>
   <addaction name="menuBatchOp"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionRaw_image">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Raw image</string>
   </property>
  </action>
  <action name="action2D_layout">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>2D layout</string>
   </property>
  </action>
  <action name="actionReset_layout">
   <property name="text">
    <string>Reset layout</string>
   </property>
  </action>
  <action name="actionLoad_image">
   <property name="text">
    <string>Load image</string>
   </property>
  </action>
  <action name="actionPredict_Layout">
   <property name="text">
    <string>Predict Layout</string>
   </property>
  </action>
  <action name="actionmergeline_save">
   <property name="text">
    <string>mergeline save</string>
   </property>
  </action>
  <action name="actionjoincorner_save">
   <property name="text">
    <string>joincorner save</string>
   </property>
  </action>
  <action name="actionlayout_textsize_save">
   <property name="text">
    <string>layout textsize save</string>
   </property>
  </action>
  <action name="actioncalline_save">
   <property name="text">
    <string>calculateline save</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
